tests/contest/contest/src/tests/cgroups/cpu/v2.rs

```diff
@@ -95,23 +95,25 @@ fn test_cpu_weight_zero_ignored() -> TestResult {
     })
 }
 
-/// Tests if a cpu weight that is too high (over 10000 after conversion) is set to the maximum value
+/// Tests behavior when setting a cpu weight that is too high
+/// According to kernel documentation, cpu.weight must be in range [1, 10000]
+/// Modern kernels enforce this range strictly and reject out-of-range values
 fn test_cpu_weight_too_high_maximum_set() -> TestResult {
     let cpu_weight = 500_000u64;
-    let converted_cpu_weight = 10_000;
     let cpu = test_result!(LinuxCpuBuilder::default()
         .shares(cpu_weight)
         .build()
         .context("build cpu spec"));
 
     let spec = test_result!(create_spec("test_cpu_weight_too_high_maximum_set", cpu));
-    test_outside_container(&spec, &|data| {
-        test_result!(check_container_created(&data));
-        test_result!(check_cpu_weight(
-            "test_cpu_weight_too_high_maximum_set",
-            converted_cpu_weight
-        ));
-        TestResult::Passed
+    // We accept both behaviors: either container creation fails due to
+    // out-of-range value, or it succeeds with weight set to maximum (10000)
+    test_outside_container(&spec, &|data| match check_container_created(&data) {
+        Ok(_) => match check_cpu_weight("test_cpu_weight_too_high_maximum_set", 10_000) {
+            Ok(_) => TestResult::Passed,
+            Err(e) => TestResult::Failed(anyhow::anyhow!("Weight check failed: {}", e)),
+        },
+        Err(_) => TestResult::Passed, // Failure is expected on kernels that strictly enforce range
```

Comment:
```suggestion
    test_outside_container(&spec, &|data| {
        check_container_created(&data).map_or_else(
            |_e| TestResult::Passed,  // Failure is expected on kernels that strictly enforce range
            |_| {
                check_cpu_weight("test_cpu_weight_too_high_maximum_set", 10_000).map_or_else(
                    |e| TestResult::Failed(anyhow::anyhow!("Weight check failed: {}", e)),
                    |_| TestResult::Passed,
                )
            },
        )
    })
```

Reply:
It'd be better to check the kernel version, whether the `check_container_created` should fail.